standard/expressions.md

```diff
@@ -4568,7 +4716,14 @@ The conditional operator is right-associative, meaning that operations are group
 
 The first operand of the `?:` operator shall be an expression that can be implicitly converted to `bool`, or an expression of a type that implements `operator true`. If neither of these requirements is satisfied, a compile-time error occurs.
 
-The second and third operands, `x` and `y`, of the `?:` operator control the type of the conditional expression. If both `x` and `y` are *default_literal*s ([§12.8.20](expressions.md#12820-default-value-expressions)), a compile-time error occurs.
+If `ref` is present:
+
+- An identity conversion must exist between the types of the two *variable_reference*s, and type of the result can be either type. If either type is `dynamic`, type inference prefers `dynamic` (§8.7).
```

Comment:
What does it mean that the type of the result can be either type?  That the compiler gets to decide?  That it is not defined which becomes the type?

Reply:
I’m all for rigor in specification, some might say pedantically so 😉, but I think I’d let this one slide.

Reply:
This is an issue not just for ref, but regular conditionals. For example:

```csharp
bool b = true;
var z = b ? (a: 1, b: 2) : (a: 3, d: 4);
```

The type of `z` is `(a: int, int)`... that needs to be specified somewhere.

Bill has created #803 for this. We need to fix this for C# 7, both for ref and non-ref.

Reply:
See https://github.com/dotnet/roslyn/blob/e006309c2c8748452b7bb404ad07271e8e029d37/src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs#L251 for the relevant compiler code.  Merging of equivalent types should be specified, but can reasonably be deferred until after v7 ships.